src/settings.js

```diff
@@ -166,31 +166,56 @@ const settings = {
     },
     /**
-     * @name settings.CorrectionRange
-     * @param {corrections} corrections e.g. { range: { min: '80', max: '150' } });
-     * corrections:{
-     *    range:{ min: string, max: string,},
-     *     preMeal:{min: string, max: string},
-     *     workout:{min: string, max: string}
-     * },
+     * @name settings.CorrectionRanges
+     * @param {object} ranges e.g. [{ time: '12:00 AM', min: '80', max: '150' }];
      */
-    async CorrectionRange(corrections) {
-        console.log('corrections: ', corrections);
+    async CorrectionRanges(ranges) {
+        const correctionRangePickerColumns = {
+            Units: 5,
+            MaximumValue: 4,
+            Separator: 3,
+            MinimumValue: 2,
+            Time: 1,
+        };
+
         await match.accessible.ButtonBarButton('Settings').tap();
         await match.accessible.Text('Correction Range').tap();
         await match.accessible.ButtonBarButton('Add').tap();
 
-        if (corrections.range) {
-            console.log('corrections.range: ', corrections.range);
-            //TODO: atIndex!! need a better way
-            // await match.accessible.Label('12:00 AM').atIndex(0).tap();
-            // await match.accessible.Label(`${corrections.range.max}`).atIndex(1).tap();
-            // await match.accessible.Label(`${corrections.range.min}`).atIndex(0).tap();
-            // await match.accessible.Label('Save').tap();
+        if (ranges) {
+            let pickerItemIndex = 1;
+            for (let index = 0; index < ranges.length; index++) {
+                const range = ranges[index];
+                await match.accessible.Label(`${range.time}`).atIndex(0).tap();
+                await match.accessible.PickerItem(pickerItemIndex, `${range.max}`).tap();
+                await match.accessible.PickerItem(pickerItemIndex, `${range.min}`).atIndex(correctionRangePickerColumns.MinimumValue).tap();
+                pickerItemIndex++;
+            }
         }
-        if (corrections.preMeal) {
-            console.log('corrections.preMeal: ', corrections.preMeal);
+        await match.accessible.Label('Save').tap();
+        await match.accessible.BackButton('Settings').tap();
+        await match.accessible.ButtonBarButton('Done').tap();
+    },
+    /**
+     * @name settings.CorrectionRangeOverride
+     * @param {object} override e.g. { min: '80', max: '150' };
+     */
+    async PreMealCorrectionRange(override) {
+        const glucoseRangeOverridePickerColumns = {
+            Label: 1,
+            MinimumValue: 2,
+            Separator: 3,
+            MaximumValue: 4,
+            Units: 5,
+        };
+        await match.accessible.ButtonBarButton('Settings').tap();
+        await match.accessible.Text('Correction Range').tap();
+        if (override) {
+            await match.accessible.Label('Pre-Meal').tap();
+            await match.accessible.PickerItem(2, `${override.max}`).tap(); //sets max
```

Comment:
[?] Does this not need `atIndex` because it is active by default? If so, might still want to set index to be more fault tolerate of code changes which change the default active field.